client/webserver/site/src/js/wallets.ts

```diff
@@ -576,6 +621,46 @@ export default class WalletsPage extends BasePage {
     window.open(url.toString())
   }
 
+  async recoverWallet () {
+    const page = this.page
+    const req = {
+      assetID: this.reconfigAsset,
+      appPW: page.recoverWalletPW.value
+    }
+    const url = '/api/recoverwallet'
+    const loaded = app().loading(page.forms)
+    const res = await postJSON(url, req)
+    loaded()
+    if (res.code === activeOrdersErrCode) {
+      this.forceUrl = url
+      this.forceReq = req
+      this.showForm(page.confirmForce)
+    } else if (app().checkResponse(res)) {
+      this.closePopups()
+    } else {
+      page.recoverWalletErr.textContent = res.msg
+      Doc.show(page.recoverWalletErr)
```

Comment:
You'll want to `Doc.hide` this at the top of `recoverWallet` too, as a lingering message for an e.g. password typo might be confusing while the request is being processed.